app/javascript/components/public_health/PatientsTable.js

```diff
@@ -272,6 +319,12 @@ class PatientsTable extends React.Component {
     });
   };
 
+  advancedFilterUpdate(filter) {
+    this.setState({ filter: filter.filter(field => field?.filterOption != null) }, () => {
```

Comment:
I'm seeing this error after clearing out a filter

![Screen Shot 2020-10-07 at 5 04 53 PM](https://user-images.githubusercontent.com/9956561/95387960-7f277480-08bf-11eb-84d0-7d6cb8af6257.png)

Reply:
Looks like a 500... can you show the console?

Reply:
![Screen Shot 2020-10-08 at 12 34 26 AM](https://user-images.githubusercontent.com/9956561/95415362-0b588c80-08fe-11eb-97ff-c7f8729a0e75.png)

Reply:
![Screen Shot 2020-10-08 at 12 42 09 AM](https://user-images.githubusercontent.com/9956561/95415744-1f50be00-08ff-11eb-9aec-9ad3b1c51697.png)

Reply:
Thanks! I think you may have a weird local state (that controller action happens on load, I think the problem is the JSON parsing) - I'm guessing from an older review (I did change how that works between your two reviews). Clear our your saved filters (`UserFilter.delete_all` in console).

Reply:
So that got rid of one of the errors but I'm still seeing this when I try to click on 'clear current filter':

![Screen Shot 2020-10-08 at 3 13 24 PM](https://user-images.githubusercontent.com/9956561/95503136-e8b78980-0978-11eb-8038-8b051f4ead89.png)

Reply:
I can't seem to reproduce this either locally or on demo. Can you provide instructions for how to reproduce?

Reply:
Resolved offline; was a local issue